pcdet/models/dense_heads/target_assigner/axis_aligned_target_assigner.py

```diff
@@ -38,24 +38,33 @@ def assign_targets(self, all_anchors, gt_boxes_with_classes):
         Args:
             all_anchors: [(N, 7), ...]
             gt_boxes: (B, M, 8)
+
+
+        Creates anchors for each class and assigns targets to them.
+        Performs fore- and background sampling/ balancing based on class labels.
+        Here we do sampling based on FDI classes, not the auxiliary classes
+
         Returns:
 
         """
 
         bbox_targets = []
         cls_labels = []
         reg_weights = []
+        aux_cls_labels = []
 
         batch_size = gt_boxes_with_classes.shape[0]
-        gt_classes = gt_boxes_with_classes[:, :, -1]
-        gt_boxes = gt_boxes_with_classes[:, :, :-1]
+        gt_classes = gt_boxes_with_classes[:, :, 7]
+        gt_aux_classes = gt_boxes_with_classes[:, :, 8]
```

Comment:
Note that this does require users to provide a auxilliary label in the gt bbox to the model, even if aux class is not predicted/ used.
Alternatively I'd have to add if-else statements everywhere which seemed like overkill. You can still train without using the aux class though if you want and can populate the aux label with zeros or something

Reply:
I don't know if a possible workaround could be to use
```
        gt_aux_classes = gt_boxes_with_classes[:, :, -1]
```

In that way the variable is always valid and if there's one less class, the content is just a repetition of the first class. In general this trick could allow for variable input sizes in this specific case of 1 or 2 labels.

Reply:
Ah yes nice idea, done